### PR TITLE
Create binaries on each release

### LINKED ISSUE #86: Binaries in releases
Hello,
Do you plan on distributing binaries with your releases? It would allow us to pull them directly from GitHub and not need to compile or build locally,
Thanks!

### PR DESCRIPTION
**Pull request by willfurnell**
This pull request adds functionality, using 'goreleaser', to build binaries on each release.

resolves #86

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      -
+        name: Set up Go
+        uses: actions/setup-go@v5
```
Comment (abhinavDhulipala) [inline .github/workflows/release.yml:25]: Please add the following to stay consistent with our test:

```yml
      with:
        go-version: '>=1.20'
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+name: goreleaser
```
Comment (abhinavDhulipala) [inline .github/workflows/release.yml:5]: To pass reuse, we must add the following header to the top of the file:

```yml
# SPDX-FileCopyrightText: 2023 Rivos Inc.
#
# SPDX-License-Identifier: Apache-2.0
```

**Review by abhinavDhulipala — CHANGES_REQUESTED** [review]

**Comment by abhinavDhulipala** [comment]
We might also want to consider making this action dependent on the testing stage of our pipeline. All though it could also be argued that it is safe to assume the code at the tag would be working, I would still put the clause in just to be sure. Our pre-commit and tests are fast anyway.

**Comment by willfurnell** [comment]
Annoyingly making it dependent on the testing stage of the pipeline looks difficult - because you _can_ do it, but then it will run every time the verify/testing action runs - and we don't want to be building binaries for every commit, I think? https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#workflow_run
Up to you though as it's your project! :)

**Comment by abhinavDhulipala** [comment]
This is awesome! Thanks a ton. I'll merge this in and give it a try.

**Review by abhinavDhulipala — APPROVED** [review]
Looks great! I'll create a new micro release in the next couple days to test this out.

**Pull request merged**